### DISCUSSION TITLE
Local document vs GitHub

### DISCUSSION THREAD

**Discussion by nareshbhatia** [six7/figma-tokens#539] [Q&A] [answered]
1. Where does "Local document" option store token data? Along with the figma file in the cloud? I guess I don't know how plugins store their data.
2. "Local document" & GitHub are currently either/or options. I have chosen the GitHub option. However, with this option, I loose my work if the plugin crashes (I think). Is it possible to have both options active simultaneously? This means that any work gets saved instantaneously in Local document. Syncing to GitHub is only on demand.
3. The earlier point also helps in publishing the figma file to the community. Currently if I publish with the GitHub option on, the tokens are not available in copies because people don't have an access token to my GitHub repo.
4. Does switching between the two options work smoothly? Are tokens transferred over without any loss? Just wanted to confirm.

**Comment by six7** [comment, marked as answer]
👋 

1. It's stored on the current document, more specifically on figma.root on the sharedPluginData namespace "tokens" with a key of "values". That means other plugins can pick up those values.
2. Yes, this is a current pain point. We'll need to work on a local storage that allows users, if the tokens differ, to choose if they want to overwrite their tokens with the github ones or keep the local changes. Here's an issue for that: https://github.com/six7/figma-tokens/issues/353
3. I'd recommend to change back to Local when you're publishing the file.
4. Yes, you can switch smoothly. It's using the same set of tokens, so going back and forth should just be changing what sync method was selected.

**Reply by nareshbhatia** [reply]
Thank you for clarifying, @six7. This will help me a lot in how I use the plugin.
